### PR TITLE
Add design doc for coverage profiling (CP-17268)

### PR DESCRIPTION
**Pull request by lindig**
The design doc describes how we plan to add coverage profiling to Oasis-controlled projects.

### CODE REVIEW

**Comment by jonludlam** [comment]
Looks really nice. It'd be good to see a description of what the screenshot is all about - perhaps it would be useful to show a screenshot with some bits that weren't covered too?

File: profiling/coverage.md [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+that sets the environment variable from inside the program. The files
+are written to a temporary directory (respecting `$TMP` or using `/tmp`)
+and uses the `string`-typed argument to include it in the name. To be
+effective, this function must be called before the programs exits. For
+clarity it is called at the begin of program execution.
+
+## Instrumenting an Oasis Project
+
+While instrumentation is easy on the level of a small file or project it
+is increasingly less so in a bigger project. We decided to focus on
```
Comment (jonludlam) [inline profiling/coverage.md]: 'increasingly less so' is a bit awkward. Perhaps 'it becomes more challenging as the project grows'?

File: _data/navbar.yml [hunk 2]
```diff
@@ -31,6 +31,7 @@
       - xen-api/metrics.md
       - xen-api/memory.md
       - xen-api/xencenter.md
+        # - profiling/coverage.md
```
Comment (johnelse) [inline _data/navbar.yml]: Did you mean to leave this commented out?

Comment (lindig) [inline _data/navbar.yml]: I wasn't sure where to put it. I had it here first before I turned it into a design doc. I'm still not sure.

File: profiling/coverage.md (hunk 1 above)
Comment (lindig) [inline profiling/coverage.md]: Thanks - will update it.

File: _data/navbar.yml (hunk 2 above)
Comment (robhoes) [inline _data/navbar.yml]: It is not needed, because you are using `design_doc: true`, which causes it to show up in the table of design docs.

Comment (lindig) [inline _data/navbar.yml]: I removed it.

**Comment by simonjbeaumont** [comment]
Looks like all comments have been addressed. Merging.

**Pull request merged**